aim/sdk/adapters/optuna.py

```diff
@@ -0,0 +1,189 @@
+import functools
+from typing import Callable, Optional, Sequence, Union
+
+import optuna
+from optuna._experimental import experimental_class, experimental_func
+from optuna._imports import try_import
+from optuna.study.study import ObjectiveFuncType
+
+
+with try_import() as _imports:
+    from aim.sdk.run import Run
+    from aim.ext.resource.configs import DEFAULT_SYSTEM_TRACKING_INT
+
+
+@experimental_class("2.9.0")
+class AimCallback:
+    """Callback to track Optuna trials with Aim.
+
+    This callback enables tracking of Optuna study in Aim.
+    The study is tracked as a single experiment run,
+    where all suggested hyperparameters and optimized metrics
+    are logged and plotted as a function of optimizer steps.
+
+    .. note::
+        Users who want to run multiple Optuna studies within the same process
```

Comment:
I guess this case should be different with `aim`. Do I understand correctly that each study will be passed it's own `AimCallback` instance?

Reply:
No, the `AimCallback` instance is the same, we just create a new `aim.Run` for each trial.
Good point, we set the run_hash to None on line 132.  Maybe the docstring should be modified as:

"...should call ``aim.experiment.close()`` and set the aim.experiment._run_hash to None between subsequent calls to..."

What do you think?

Reply:
I don't think that users should access the private properties of the callback, maybe provide `close()` method on the callback itself, which will close the run, and reset the run hash, and mention that one in the docstring